tsfile/src/main/java/org/apache/iotdb/tsfile/encoding/decoder/PlainDecoder.java

```diff
@@ -54,26 +55,55 @@ public boolean readBoolean(ByteBuffer buffer) {
 
   @Override
   public short readShort(ByteBuffer buffer) {
+    if (this.endianType == EndianType.LITTLE_ENDIAN) {
+      int ch1 = ReadWriteIOUtils.read(buffer);
+      int ch2 = ReadWriteIOUtils.read(buffer);
+      return (short) (ch1 + (ch2 << 8));
+    }
     return buffer.getShort();
```

Comment:
A better way would be setting the endian of the buffer using `byteBuffer.order()` when the buffer is created, which can avoid many redundant checks.
Notice: ByteBuffers get from methods like `byteBuffer.slice()` do not inherit the byte order, so you have to set byte order for those ByteBuffers.

Reply:
> A better way would be setting the endian of the buffer using `byteBuffer.order()` when the buffer is created, which can avoid many redundant checks.
> Notice: ByteBuffers get from methods like `byteBuffer.slice()` do not inherit the byte order, so you have to set byte order for those ByteBuffers.

This is a very good point! Thank you.